src/Features/CSharp/Portable/Completion/CompletionProviders/SnippetCompletionProvider.cs

```diff
@@ -97,8 +97,11 @@ public override async Task ProvideCompletionsAsync(CompletionContext context)
             var context = await completionContext.GetSyntaxContextWithExistingSpeculativeModelAsync(document, cancellationToken).ConfigureAwait(false);
             var semanticModel = context.SemanticModel;
 
-            if (context.IsInTaskLikeTypeContext)
+            if (context.IsInTaskLikeTypeContext ||
+                context.IsGenericConstraintContext)
```

Comment:
⚠️  This is going to be really annoying for a user who created a custom snippet for a commonly-used type. I would prefer to leave the current behavior unchanged (snippets frequently show up in broader context than applicable), and fix it as part of a future change to semantic snippets.